### PR TITLE
PYTHON-4321 Add support for Python 3.13 and update changelog for 1.11.0 release

### PR DESCRIPTION
**Pull request by blink1073**

### CODE REVIEW

File: bindings/python/pyproject.toml [hunk 1]
```diff
@@ -36,6 +36,7 @@ classifiers = [
     "Programming Language :: Python :: 3.10",
     "Programming Language :: Python :: 3.11",
     "Programming Language :: Python :: 3.12",
+    "Programming Language :: Python :: 3.13",
```
Comment (NoahStapp) [inline bindings/python/pyproject.toml:39]: Is this really the only change we need to make to support 3.13? Wow!

**Comment by ShaneHarvey** [comment]
Are we testing Python 3.13 on evergreen?

**Comment by ShaneHarvey** [comment]
Also are the python test failures on Evergreen expected?

**Comment by blink1073** [comment]
One set of failures is fixed by https://github.com/mongodb-labs/drivers-evergreen-tools/pull/487.  I'm not sure why the one test is failing, I've yet to see a difference between this PR and https://spruce.mongodb.com/task/mongo_python_driver_tests_python_version_rhel8_test_encryption__platform~rhel8_auth_ssl~noauth_nossl_python_version~3.12_encryption~encryption_crypt_shared_test_8.0_sharded_cluster_4e74c8274e7c4cb7658d445d526dcc33ced1750b_24_09_04_12_58_14/logs?execution=0.

**Comment by ShaneHarvey** [comment]
Still think we should add 3.13 testing on evergreen if possible.

**Comment by blink1073** [comment]
> Still think we should add 3.13 testing on evergreen if possible.

Yep, I have that done locally, I'm running patch builds to figure out the test failure.

File: bindings/python/pyproject.toml (hunk 1 above)
Comment (blink1073) [reply bindings/python/pyproject.toml:39]: We had been testing in GitHub Actions, but now in EVG as well.  :smile:

**Comment by blink1073** [comment]
Okay, I found the issue: we need to use the same version of crypt_shared, the one downloaded during `run-orchestration.sh`.

**Review by NoahStapp — APPROVED** [review]

**Review by ShaneHarvey — CHANGES_REQUESTED** [review]
Still need to add 3.13 here: https://github.com/mongodb/libmongocrypt/blob/master/bindings/python/.evergreen/test.sh

**Comment by blink1073** [comment]
> Still need to add 3.13 here: [master/bindings/python/.evergreen/test.sh](https://github.com/mongodb/libmongocrypt/blob/master/bindings/python/.evergreen/test.sh?rgh-link-date=2024-09-05T21%3A18%3A27Z)

Done

**Comment by ShaneHarvey** [comment]
Note the test_fork failure is tracked by https://jira.mongodb.org/browse/PYTHON-4738.

**Review by ShaneHarvey — APPROVED** [review]

**Pull request merged**